ee/clickhouse/views/experiments.py

```diff
@@ -0,0 +1,84 @@
+from typing import Any
+
+from rest_framework import request, serializers, viewsets
+from rest_framework.decorators import action
+from rest_framework.exceptions import ValidationError
+from rest_framework.permissions import IsAuthenticated
+from rest_framework.request import Request
+from rest_framework.response import Response
+
+from ee.clickhouse.queries.experiments.funnel_experiment_result import ClickhouseFunnelExperimentResult
+from posthog.api.routing import StructuredViewSetMixin
+from posthog.models.experiment import Experiment
+from posthog.models.feature_flag import FeatureFlag
+from posthog.models.filters.filter import Filter
+from posthog.models.team import Team
+from posthog.permissions import ProjectMembershipNecessaryPermissions, TeamMemberAccessPermission
+
+
+class ExperimentSerializer(serializers.ModelSerializer):
+    class Meta:
+        model = Experiment
+        fields = [
+            "id",
+            "name",
+            "description",
+            "start_date",
+            "end_date",
+            "feature_flag",
+            "parameters",
+            "filters",
+            "created_by",
+            "created_at",
+            "updated_at",
+        ]
+        read_only_fields = [
+            "id",
+            "created_by",
+            "created_at",
+            "updated_at",
+        ]
+
+    def create(self, validated_data: dict, *args: Any, **kwargs: Any) -> Experiment:
+        request = self.context["request"]
+        validated_data["created_by"] = request.user
+        team = Team.objects.get(id=self.context["team_id"])
+
+        # feature_flag = FeatureFlag.objects.filter(key=validated_data["feature_flag"], team_id=self.context["team_id"], deleted=False).first()
```

Comment:
I'll now work on this @liyiy : 

Such that, you pass in the FF key (not primary key), and if it exists, I update the FF. If it doesn't, I'll create a multivariate one with the appropriate settings.

Reply:
And at this point I guess we know enough to actually get things in. Don't want to keep adding to this PR because it's getting too huge, and we can add any extra fields to the model later on, if we still ended up missing something.

Reply:
lemme know when you're ready for this to be looked over/approved (since it's still in draft mode rn) 👍

Reply:
Still fixing some annoying edge cases with FF updates. Your FE bit remains broken without it (sigh), so will fix before requesting approval (should be tomorrow! :) )

Reply:
yep npnp, plenty of work for me to do still on the FE even without this yet 😁